paper/src/main/java/com/denizenscript/denizen/paper/events/EntityStepsOnScriptEvent.java

```diff
@@ -0,0 +1,108 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizen.utilities.Utilities;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import io.papermc.paper.event.entity.EntityMoveEvent;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EntityStepsOnScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // entity steps on block
+    // <entity> steps on <material>
+    //
+    // @Location true
+    //
+    // @Group Paper
+    //
+    // @Plugin Paper
+    //
+    // @Warning This event may fire very rapidly.
+    //
+    // @Cancellable true
+    //
+    // @Triggers when an entity steps onto a material.
+    //
+    // @Context
+    // <context.entity> returns an EntityTag of the entity stepping onto the block.
+    // <context.location> returns a LocationTag of the block the entity is stepping on.
+    // <context.previous_location> returns a LocationTag of where the entity was before stepping onto the block.
+    // <context.new_location> returns a LocationTag of where the entity is now.
+    //
+    // @Example
+    // # Announce the name of the entity stepping on the block and the material of block.
+    // on entity steps on block:
+    //     - announce "<context.entity.name> stepped on a <context.location.material.name>!"
+    //
+    // @Example
+    // # Announce the material of the block a sheep has stepped on.
+    // on sheep steps on block:
+    //     - announce "A sheep has stepped on a <context.location.material.name>!"
+    //
+    // @Example
+    // # Announce that a sheep has stepped on a diamond block.
+    // on sheep steps on diamond_block:
+    //     - announce "A sheep has stepped on a diamond block! Must be a wealthy sheep!"
+    // -->
+
+    public EntityStepsOnScriptEvent() {
+        registerCouldMatcher("<entity> steps on <material>");
+    }
+
+    public EntityTag entity;
+    public LocationTag location;
+    public LocationTag previousLocation;
+    public LocationTag newLocation;
+    public EntityMoveEvent event;

```

Comment:
You keep the event instance but don't seem to actually use it, should either remove that field or replace some of the existing ones with directly reading the data from the event